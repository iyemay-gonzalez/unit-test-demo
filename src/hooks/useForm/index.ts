import useForm from './useForm.hook';

export { useForm };
