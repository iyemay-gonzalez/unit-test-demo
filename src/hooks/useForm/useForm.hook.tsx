import React, {ChangeEvent, useContext, useEffect, useState} from 'react';
import InUseForm, {
	InUseFormReturn,
} from './useForm.types';
import { LoginFormValues } from "../../components/loginComponent/loginForm.types";


const useForm = ({
     initialValues,
     onSubmit,
 }: InUseForm): InUseFormReturn => {
	const [state, setState] = useState<LoginFormValues>({
		userName: initialValues.userName,
		userPassword: initialValues.userPassword,
	});
	
	const handleChange = (event: ChangeEvent<HTMLInputElement | HTMLSelectElement>): void => {
		const { target } = event;
		const name = target.name;
		const value = target.value;
		setState(prevState => ({
			...prevState,
			[name]: value
		}));
	};
	
	const handleSubmit = (event: ChangeEvent<HTMLFormElement>): void => {
		event.preventDefault();
		onSubmit(state);
	};
	
	return {
		formValues: state,
		handleChange,
		handleSubmit,
	};
}

export default useForm;