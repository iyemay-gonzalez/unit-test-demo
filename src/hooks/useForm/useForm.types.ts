import { LoginFormValues } from "../../components/loginComponent/loginForm.types";

interface InUseForm {
	initialValues: LoginFormValues;
	onSubmit: Function;
}

export interface InUseFormReturn {
	formValues: LoginFormValues;
	handleChange: (
		event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>,
	) => void;
	handleSubmit: (
		event: React.ChangeEvent<HTMLFormElement>
	) => void;
}

declare const InUseForm: InUseForm;

export default InUseForm;