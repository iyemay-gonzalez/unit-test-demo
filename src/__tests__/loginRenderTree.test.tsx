import { expect, test } from "@jest/globals";
import { create } from "react-test-renderer";
import LoginForm from "../components/loginComponent/loginForm.component";
import { renderHook } from '@testing-library/react-hooks';
import {useForm} from "../hooks/useForm";
import LoginRequest from "../request/loginRequest";
import {LoginFormValues} from "../components/loginComponent/loginForm.types";
import {initialValues} from "../components/loginComponent/formLogin.state";
import {build, fake} from "@jackfranklin/test-data-bot";

interface Form {
	userName: string;
	userPassword: string;
}

const buildForm = build<Form>({
	fields: {
		userName: fake(f => f.name.findName()),
		userPassword: 'Pass123@-%@2',
	},
});

const {
	userName,
	userPassword,
} = buildForm();

const newInitialValues: LoginFormValues = {
	...initialValues,
	userName: userName,
	userPassword: userPassword,
};

const { result } = renderHook(() =>
	useForm({
		initialValues: newInitialValues,
		onSubmit: LoginRequest,
	}),
);

test("renders correctly with no pets", () => {
	const tree = create(
		<LoginForm
			formValues={result.current.formValues}
            handleChange={result.current.handleChange}
            handleSubmit={result.current.handleSubmit}
		/>
	).toJSON();
	expect(tree).toMatchSnapshot();
});