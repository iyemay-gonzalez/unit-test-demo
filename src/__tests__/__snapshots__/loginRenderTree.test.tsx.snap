// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with no pets 1`] = `
<div
  className="sc-bdvvtL lbtdOu"
>
  <div
    className="sc-gsDKAQ hFhazj"
  >
    <div
      className="sc-dkPtRN kUjrfe"
    >
      <h1
        className="card-title"
      >
        Pruebas Unitarias
      </h1>
    </div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="sc-hKwDye dlqTEB"
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="exampleInputName"
          >
            Usuario
          </label>
          <input
            className="form-control"
            data-testid="userName"
            id="exampleInputName"
            name="userName"
            onChange={[Function]}
            type="name"
            value="Shaun Mante"
          />
        </div>
      </div>
      <div
        className="sc-hKwDye dlqTEB"
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="exampleInputPassword"
          >
            Contraseña
          </label>
          <input
            className="form-control"
            data-testid="userPass"
            id="exampleInputPassword"
            name="userPassword"
            onChange={[Function]}
            type="password"
            value="Pass123@-%@2"
          />
        </div>
      </div>
      <div
        className="sc-eCImPb ewExDR"
      >
        <button
          className="btn btn-primary"
          data-testid="buttonUser"
          type="submit"
        >
          Iniciar Sesión
        </button>
      </div>
    </form>
  </div>
</div>
`;
